src/ApiCompat/Microsoft.DotNet.ApiCompat.Tool/Program.cs

```diff
@@ -368,7 +369,7 @@ private static (string CaptureGroupPattern, string ReplacementString)[]? ParseTr
             Dictionary<NuGetFramework, IEnumerable<string>> packageAssemblyReferencesDict = new(argumentResult.Tokens.Count);
             foreach (var token in argumentResult.Tokens)
             {
-                string[] parts = token.Value.Split('=');
+                string[] parts = token.Value.Split(';');
```

Comment:
That fixes a bug introduced with https://github.com/dotnet/sdk/commit/b8c07fee1b150d46e5c4229cf1f0d81da20393a5. The expected input changed from the TFM simple name to the moniker, i.e. `net7.0` became `.NETCoreApp,Version=v7.0`. We can't use the `=` character as a delimiter here anymore as that's used by the moniker string. The semicolon seems safe to use.

Overall I don't expect many people to use the package validation option in the CLI frontend and even less often to pass in package assembly references. Because of that, I'm not worried about the weird argument syntax.

Reply:
The `;` character is normally used to indicate the end of a command and to start concatenating another command in the CLI.

How about using `,` instead?

Reply:
Also what's the name of the argument to which the user would pass the list of tokens? This separator should be documented.

Reply:
> This separator should be documented.

We are way behind documentation. We didn't document the previous delimiter either. That's tracked via https://github.com/dotnet/sdk/issues/27857

> The ; character is normally used to indicate the end of a command and to start concatenating another command in the CLI.

👍 

> Also what's the name of the argument to which the user would pass the list of tokens?

`--package-assembly-references`

> How about using , instead?

`,` is an expected value in the `TargetFrameworkMoniker` property: `.NETCoreApp,Version=v7.0`.

Reply:
I see.

If you do choose to pick `;` as the separator, then the whole argument needs to be enclosed with double quotes to prevent the CLI to think you're separating commands with it. I think you could test this in PowerShell and confirm if my suspicion is correct.

The docs can wait for now.

Maybe another separator character could be `&`.

Reply:
Changed to `|` which doesn't have a special meaning when surrounded with other characters. Tested it with surrounding quotes and without.